Source/JavaScriptCore/API/JSClassRef.cpp

```diff
@@ -51,14 +51,14 @@ OpaqueJSClass::OpaqueJSClass(const JSClassDefinition* definition, OpaqueJSClass*
     , callAsConstructor(definition->callAsConstructor)
     , hasInstance(definition->hasInstance)
     , convertToType(definition->convertToType)
-    , m_className(String::fromUTF8(definition->className))
+    , m_className(String::fromUTF8(unsafeNullTerminated(definition->className)))
```

Comment:
I'm unclear why we don't just use `String::fromUTF8(unsafeSpan(definition->className))` for cases like these. Why do we need yet another type?

Reply:
In this case (and cases like it that construct String or StringView), unsafeNullTerminated and unsafeSpan have equivalent results. I chose unsafeNullTerminated because it more closely conveys what we know at the call site, and what we'll eventually be able to annotate with the compiler. But if we prefer unsafeSpan for now, that preference seems fine to me too.

The case where you _need_ a new type the most is when you invoke a `char*` API that requires null-termination. For example, SandboxExtensionImpl::create.

Reply:
Are there safer cpp warnings in sandbox extension code? I did not think we had any left.